src/Components/Server/src/Prerendering/ComponentPrerenderResult.cs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.IO;
+
+namespace Microsoft.AspNetCore.Components.Server.Prerendering
+{
+    /// <summary>
+    /// Represents the result of a prerendering an <see cref="IComponent"/>.
+    /// </summary>
+    public sealed class ComponentPrerenderResult
```

Comment:
I plan to lift this into Microsoft.AspNetCore.Components so that I can pass in the RenderTreeFrames and perform the serialization into a stream/pipe directly from here instead of relying on the internal list of tokens.

Reply:
Remember that any reads you do from `RenderTreeFrames` must be synchronous, and hence can't be interleaved with network I/O. This is because `RenderTreeFrames` is a mutable structure, and we want to keep it that way.

You could arguably copy its binary data into a separate buffer if you want, though whether that's better than doing the conversion-to-HTML-string up front is unclear.

Reply:
Yes, I don't expect this to be much of a problem. The general recommendation would be to consume the results right away, and I have ideas on how to further prevent more renders after we decide we are done with prerendering and until we reconect to the server, so the render tree frames shouldn't change after we are done prerendering for the remainder of the request.